Verify that the snapshot actually contains descendant data

jsf9k opened the pull request:
## 🗣 Description ##

This pull request fixes an issue where report generation can fail if _all_ the descendants of an org have start dates in the future (i.e., are not yet scanning).

## 💭 Motivation and context ##

Resolves #92.

## 🧪 Testing ##

I made this code change _en vivo_ on our production system and then verified that I was able to successfully create the offending report from #92.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

jsf9k committed:
Verify that the snapshot actually contains descendant data

If _all_ the descendants of an org have start dates in the
future (i.e., are not yet scanning) then
self.__snapshots[0].get("descendants_included") will be true but
"ss0_descendant_data" will not be a key of self.__results.  We need to
avoid this corner case.

Resolves issue #92.

dav3r commented on cyhy_report/customer/generate_report.py:
```diff
@@ -2947,7 +2947,14 @@ def __generate_false_positives_attachment(self):
+        # self.__results.  We need to avoid this corner case.
+        #
+        # See issue #92 for more details.
+        if self.__snapshots[0].get("descendants_included") and "ss0_descendant_data" in self.__results:
```
Now that I have thought about it, this is a little redundant.  I don't think there can be a situation where `"ss0_descendant_data" in self.__results` is true and `self.__snapshots[0].get("descendants_included")` is false, which means that this can be simplified to:
```suggestion
        if "ss0_descendant_data" in self.__results:
```

Thoughts?

jsf9k replied:
I'm fine with this.  What do you say, @felddy?

dav3r replied:
Pretty sure it's all good, since the only way that `"ss0_descendant_data"` gets added is if there are descendant snapshots:
https://github.com/cisagov/cyhy-reports/blob/63b896a2e4cfe946b446cbfe034888a2e8a4fe8a/cyhy_report/customer/generate_report.py#L826-L859

jsf9k replied:
If @felddy doesn't answer by COB tomorrow then I'm fine proceeding without his imprimatur.

jsf9k replied:
We shall proceed without @felddy's imprimatur.

jsf9k committed:
Simplify logic in if clause

@dav3r pointed out that there can never be a situation where "ss0_descendant_data" in self.__results is true and self.__snapshots[0].get("descendants_included") is false.  Thus the self.__snapshots[0].get("descendants_included") check is redundant.

Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>

dav3r reviewed (approved):
👍 👍  Thank you for squashing this bug!
